### ISSUE TITLE
Make SDK easily compileable to GraalVM native image

### ISSUE THREAD

**Issue by gustf** [aws/aws-sdk-java-v2#1852]
This is a repost of this issue in v1, very much wanted in v2 also

https://github.com/aws/aws-sdk-java/issues/2037

**Comment by debora-ito** [comment]
Marking as a feature request. 

If you want to show support for it please +1 the original post, it helps us with prioritization.

**Comment by Vinaybs** [comment]
+1

**Comment by kiransterling** [comment]
+1

**Comment by wtsiamruk** [comment]
+1

**Comment by hendrikhalkow** [comment]
+10

**Comment by wtsiamruk** [comment]
+100

**Comment by wtsiamruk** [comment]
moreover, quakus team already done that in some way https://github.com/aws-samples/aws-quarkus-demo, maybe some collaboration can be done here? https://github.com/aws-samples/aws-quarkus-demo

**Comment by zoewangg** [comment]
Hey all, we have added support for GraalVM native image in the SDK via #2268, and the change has been released in `2.16.1`.  You can checkout a sample native image application here https://github.com/aws/aws-sdk-java-v2/tree/master/test/sdk-native-image-test

We also relased a new maven archetype, https://github.com/aws/aws-sdk-java-v2/tree/master/archetypes/archetype-app-quickstart, which enables you to quickly bootstrap a simple application with built-in native image support and SDK 2.x as a dependency

Example command:
```
mvn archetype:generate \
    -DarchetypeGroupId=software.amazon.awssdk \
    -DarchetypeArtifactId=archetype-app-quickstart \
    -DarchetypeVersion=2.16.1 \
    -DgroupId=com.test \
    -DartifactId=sample-project \
    -Dservice=s3  \
    -DhttpClient=apache-client \
    -DnativeImage=true \
    -Dregion=us-west-2 \
    -DinteractiveMode=false
```
Please let us know if you have any feedback! 🙂
